python/py-cryptography/Portfile

```diff
@@ -1,49 +1,167 @@
-long_description    ${description}
+description         cryptography is a package designed to expose \
+                    cryptographic primitives and recipes to Python developers.
 
-homepage            https://github.com/pyca/cryptography
+long_description    {*}${description}
 
-checksums           rmd160  61fb86fd231710d5875428a3a1cefcd5c1d0c99e \
+checksums           ${distname}${extract.suffix} \
+                    rmd160  f579f749a23d076f7204bd6cf66d8dceb2ab3866 \
+                    sha256  9eb1a420f9f50e131f451b9dc8950c4f3b2a1298ee9d6e25ba873d3f66354a80 \
+                    size    35737283
+
+# https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst
+set cryptography_min_ver 14
+
+if {${name} ne ${subport}
+    && [string match "py*-${github.project}" ${subport}]} {
+    depends_build-append \
+                port:py${python.version}-setuptools
+
+    depends_lib-append \
+                path:lib/libssl.dylib:openssl \
+                port:py${python.version}-cffi \
+                port:py${python.version}-six
+
+    # legacy support
+    if {${python.version} eq 27
+        || ${os.platform} eq "darwin" && ${os.major} < ${cryptography_min_ver}} {
+        version     2.9.2
+
+        description Legacy support of Python cryptography.
+
+        long_description    {*}${description}
+
+        master_sites \
+                    pypi:[string index ${python.rootname} 0]/${python.rootname}
+        distname    ${python.rootname}-${version}
+
+        checksums   rmd160  61fb86fd231710d5875428a3a1cefcd5c1d0c99e \
                     sha256  a0c30272fb4ddda5f5ffc1089d7405b7a71b0b0f51993cb4e5dbb4590b2fc229 \
                     size    517571
 
-platform macosx {
-    if {${os.major} < 11} {
-        # https://trac.macports.org/ticket/54519
-        # https://trac.macports.org/ticket/54753
-        # https://trac.macports.org/ticket/55079
-        patchfiles-append   patch-src__cffi_src_build_openssl.py.diff
+        platform macosx {
+            if {${os.major} < 11} {
+                # https://trac.macports.org/ticket/54519
+                # https://trac.macports.org/ticket/54753
+                # https://trac.macports.org/ticket/55079
+                patchfiles-append   patch-src__cffi_src_build_openssl.py.diff
+            }
+        }
+
+        depends_lib-append \
+                port:py${python.version}-asn1crypto
+
+        if {${python.version} eq 27} {
+            depends_lib-append \
+                port:py${python.version}-enum34 \
+                port:py${python.version}-ipaddress
+        }
+
+        notes-append \
+            "This is a legacy version of ${python.rootname} that contains\
+            CVE CVE-2020-36242. Please install port\
+            py${python.version}-${python.rootname} for an up-to-date version."
+    } else {
+        PortGroup   cargo_fetch 1.0
+
+        depends_build-append \
+                port:py${python.version}-setuptools-rust \
```

Comment:
add `py-toml` here

Reply:
I think toml should be a lib dependency of setuptools-rust, which will fix this issue here and other ports that use it.

Reply:
if indeed *all* ports that use `setuptools-rust` need also `toml`, then yes that could be a solution. 

It seems a bit similar though to the `setuptools_scm` situation, where sometimes packages specify the dependency as `setuptools_scm[toml]` indicating that `toml` is also needed. The question is then where to add the `toml` dependency; I would say explicitly in the packages that need it, and not in `setuptools_scm` (or `setuptools-rust`), but I can see both sides and don't really care.

Reply:
I included a commit to setuptools-rust because I recall this issue in the tokenizers port.